src/main/java/www/purple/mixxy/controllers/ComicController.java

```diff
@@ -17,48 +26,187 @@
+		
+		Comic comic = null;
+
+		if (slug != null) {
+
+			comic = comicDao.getComic(user, slug);
+
+		}
+		
+		List <Comic> remixes = null;
+
+		if (comic.id != null) {
+
+			remixes = comicDao.getRemixes(comic.id);
+
+		}
+
+		return Results.html().render("remix", remixes);
+
+	}
 	
-  public Result remix() {
-    return Results.TODO();
-  }
-
-  public Result edit() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result newWork() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result likes() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result remixes() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result root() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result parent() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result delete() {
-    return Results.TODO();
-  }
-
-  @FilterWith(JsonEndpoint.class)
-  public Result update() {
-    return Results.TODO();
-  }
+	/**
+	 * This method creates a new remix for a specific comic.
+	 * 
+	 * Pass in the id of the parent comic to remix on along with the current user, context and data model information.
+	 * 
+	 * @param id Identifier of the Parent comic
+	 * @param username Current user's username
+	 * @param context html context
+	 * @param comicDto Data model for the comic object
+	 * @param validation Checks for various violations such as:  field violations (on controller method fields), 
+	 * bean violations (on an injected beans field) or general violations (deprecated)
+	 * 
+	 * @return resulting route to redirect with content
+	 */
+	@FilterWith(JsonEndpoint.class)
+	public Result newRemix(@PathParam("user") String user, @PathParam("work") String slug, @LoggedInUser String username, Context context,
+			@JSR303Validation ComicDto comicDto, Validation validation) {
+
+		if (validation.hasViolations()) {
+
+			context.getFlashScope().error("Please correct field.");
+			context.getFlashScope().put("title", comicDto.title);
+			context.getFlashScope().put("description", comicDto.description);
+			context.getFlashScope().put("images", comicDto.images);
+			context.getFlashScope().put("likes", comicDto.likes);
+			context.getFlashScope().put("tags", comicDto.tags);
+
+			return Results.redirect("/remix/");
+
+		} else {
+
+			Comic comic = null;
+
+			if (slug != null) {
+
+				comic = comicDao.getComic(username, slug);
+
+			}
+			
+			comicDao.branchComic(username, comic.id);
+
+			context.getFlashScope().success("New remix created.");
+
+			return Results.redirect("/");
+
+		}
+
+	}
+	
+	@FilterWith(JsonEndpoint.class)
+	public Result comicShow(@PathParam("user") String username, @PathParam("comic-slug") String slug) {
+
+		Comic comic = null;
+
+		if (slug != null) {
+
+			comic = comicDao.getComic(username, slug);
+
+		}
+
+		return Results.html().render("comic", comic);
```

Comment:
If the `comic` is not found (i.e. `null`), this may throw an exception.  If the `comic` is null, you can just return a 404 (forgot the exact method).